client/extensions/belgaCoverage/src/belga-coverage-carousel.tsx

```diff
@@ -0,0 +1,112 @@
+import * as React from 'react';
+
+import {ISuperdesk} from 'superdesk-api';
+import {Badge, Carousel, IconButton} from 'superdesk-ui-framework';
+import {IBelgaCoverage, getCoverageInfo, getCoverageImages, IBelgaImage} from './belga-image-api';
+import 'superdesk-ui-framework/dist/superdesk-ui.bundle.css';
+
+interface IProps {
+    preview?: boolean;
+    coverageId: string;
+    removeCoverage?: () => void;
+    superdesk: ISuperdesk;
+}
+
+interface IState {
+    loading: boolean;
+    coverage: IBelgaCoverage | null;
+    images: Array<IBelgaImage> | null;
+}
+
+export default class BelgaCoverageCarousel extends React.PureComponent<IProps, IState> {
+    constructor(props: IProps) {
+        super(props);
+
+        this.state = {
+            loading: true,
+            coverage: null,
+            images: null,
+        };
+    }
+
+    componentDidMount() {
+        Promise.all([
+            getCoverageInfo(this.props.coverageId),
+            getCoverageImages(this.props.coverageId, 8),
+        ]).then(([coverage, images]) => {
+            this.setState({
+                loading: false,
+                coverage,
+                images,
+            });
+        }).catch((err) => {
+            console.error(err);
+            this.setState({
+                loading: false,
+            });
+        });
+    }
+
+    render() {
+        if (this.state.loading) {
+            return (
+                <div className="sd-loader" />
+            );
+        }
+
+        const {Alert} = this.props.superdesk.components;
+        const {formatDate} = this.props.superdesk.localization;
+
+        if (this.state.coverage == null || this.state.images == null) {
+            return (
+                <Alert type="error">{'There was an error when fetching coverage ' + this.props.coverageId}</Alert>
+            );
+        }
+
+        const {preview} = this.props;
+        const {coverage, images} = this.state;
+
+
+        const itemTemplate = (props: IBelgaImage) => (
+            <div className="sd-thumb-carousel__item">
+                <div className="sd-thumb-carousel__item-inner">
+                    <img src={props.thumbnailUrl} alt={props.name} />
+                </div>
+            </div>
+        );
+
+        const headerTemplate = (
+            <div className="sd-thumb-carousel__header">
+                <h4 className="sd-thumb-carousel__heading">{coverage.name}</h4>
+                <Badge text={'' + coverage.nrImages} type='light' />
```

Comment:
Is `nrImages` image count? It'd be good to rename it or at least add a comment in the interface if the name comes from their API.